platform/ethereum/domain.go

```diff
@@ -1,14 +1,33 @@
 package ethereum
 
 import (
-	CoinType "github.com/trustwallet/blockatlas/coin"
+	"strings"
+
+	"github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/address"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/blockatlas/pkg/errors"
 	"github.com/trustwallet/blockatlas/pkg/logger"
 	"github.com/trustwallet/blockatlas/platform/ethereum/ens"
-	AddressEncoder "github.com/trustwallet/ens-coincodec"
+	coincodec "github.com/trustwallet/ens-coincodec"
 )
 
+var domains = map[string]interface{}{
+	".eth":  nil,
+	".xyz":  nil,
+	".luxe": nil,
+	".kred": nil,
+}
+
+func (p *Platform) Match(name string) bool {
+	tld := address.GetTLD(name, ".")
+	if len(tld) == 0 {
+		return false
+	}
+	_, ok := domains[strings.ToLower(tld)]
```

Comment:
It is not a good idea to use maps, use constants

Reply:
See above.  Done with set now.  Constants don't have an easy way to check against all.